File: project-meli/products-page/src/types/Product.ts
export interface ProductImage {
  urls: string[];
  mainImage: string;
}

export interface SalesInfo {
  condition: string;
  soldCount: number;
}

export interface Pricing {
  originalPrice: number;
  currentPrice: number;
  discountPercent: string;
}

export interface PaymentMethods {
  pix: {
    discountedPrice: number;
    text: string;
  };
  installments: {
    pricePerInstallment: number;
    numberOfInstallments: number;
    totalPrice: number;
    text: string;
  };
}

export interface ShippingInfo {
  deliveryTime: string;
  deliveryEstimate: string;
  pickupOption: string;
  viewMapLink: string;
}

export interface SellerInfo {
  name: string;
  rating: string;
  reviewsCount: string;
  salesCount: string;
  reputationColor: string;
  link: string;
}

export interface Feature {
  name: string;
  value: string;
}

export interface ReviewBreakdown {
  stars: number;
  percentage: number;
}

export interface ReviewComment {
  id: string;
  user: string;
  rating: number;
  date: string;
  comment: string;
}

export interface Reviews {
  average: number;
  total: number;
  breakdown: ReviewBreakdown[];
  reviewComments: ReviewComment[];
}

export interface Recommendation {
  id: string;
  name: string;
  price: number;
  imageUrl: string;
}

export interface Product {
  id: string;
  title: string;
  salesInfo: SalesInfo;
  pricing: Pricing;
  paymentMethods: PaymentMethods;
  brand: string;
  stockStatus: string;
  shippingInfo: ShippingInfo;
  productImages: ProductImage;
  sellerInfo: SellerInfo;
  features: Feature[];
  description: string;
  reviews: Reviews;
  recommendations: Recommendation[];
}

File: project-meli/products-spring-boot/src/main/java/br/com/meli/product/response/Product.java
package br.com.meli.product.response;

import java.util.List;

import br.com.meli.product.data.FeatureProduct;
import br.com.meli.product.data.PaymentMethods;
import br.com.meli.product.data.Pricing;
import br.com.meli.product.data.ProductImages;
import br.com.meli.product.data.Recommendation;
import br.com.meli.product.data.Reviews;
import br.com.meli.product.data.SalesInfo;
import br.com.meli.product.data.SellerInfo;
import br.com.meli.product.data.ShippingInfo;
public record Product(
	    String id,
	    String title,
	    SalesInfo salesInfo,       
	    Pricing pricing,            
	    PaymentMethods paymentMethods, 
	    String brand,
	    String stockStatus,
	    ShippingInfo shippingInfo,  
	    ProductImages  productImages,
	    SellerInfo sellerInfo,      
	    List<FeatureProduct> features,   
	    String description,
	    Reviews reviews,            
	    List<Recommendation> recommendations 
	) {

}
